src/main/java/de/unijena/cheminf/mortar/gui/controls/ToggleSwitch.java

```diff
@@ -0,0 +1,309 @@
+        this.switchTransition = new ParallelTransition(this.switchAnimation, this.fillAnimation);
+        this.switchAnimation.setNode(this.switchButton);
+        this.fillAnimation.setShape(this.switchBackground);
+        this.getChildren().addAll(this.switchBackground, this.switchButton);
+        //Listener
+        this.switchStateBooleanProperty.addListener((observable, oldValue, newValue) -> {
+            boolean tmpIsOn = newValue.booleanValue();
+            this.switchAnimation.setToX(tmpIsOn ? (aRectangleWidth - (2 * aCircleRadius)) : 0); //?
+            this.fillAnimation.setFromValue(tmpIsOn ? aRectangleColorOff : aRectangleColorOn);
+            this.fillAnimation.setToValue(tmpIsOn ? aRectangleColorOn : aRectangleColorOff);
+            this.switchTransition.play();
+        });
+        //Mouse listener.
+        this.setOnMouseClicked(event -> this.switchStateBooleanProperty.set(!this.switchStateBooleanProperty.get()));
+    }
+    //</editor-fold>
+    //
+    //<editor-fold desc="properties" defaultstate="collapsed">
+    /**
+     * returns switchButton.
+     *
+     * @return Circle
+     */
+    public Circle getSwitchButton() {
+        return this.switchButton;
+    }
+    /**
+     * returns switchBackground.
+     *
+     * @return Rectangle
+     */
+    public Rectangle getSwitchBackground() {
+        return this.switchBackground;
+    }
+    /**
+     * returns switchAnimation which shows the visual transition of the Circle switchButton.
+     *
+     * @return TranslateTransition
+     */
+    public TranslateTransition getSwitchAnimation() {
+        return this.switchAnimation;
+    }
+    /**
+     * returns fillAnimation to show the color change when clicked on.
+     *
+     * @return FillTransition
+     */
+    public FillTransition getFillAnimation() {
+        return this.fillAnimation;
+    }
+    /**
+     * returns switchTransition to maintain a parallel animation for fillAnimation and switchAnimation.
+     *
+     * @return ParallelTransition
+     */
+    public ParallelTransition getSwitchTransition() {
+        return this.switchTransition;
+    }
+    /**
+     * returns isSwitchedOn, the current boolean state of the Toggleswitch,
+     * if it is turned on it returns true, otherwise false.
+     *
+     * @return switch value
+     */
+    public boolean getSwitchStateBooleanProperty() {
+        return this.switchStateBooleanProperty.get();
+    }
+    /**
+     * sets switchStateBooleanProperty to update new value.
+     *
+     * @param switchStateBooleanProperty boolean
+     */
+    public void setSwitchStateBooleanProperty(boolean switchStateBooleanProperty) {
+        this.switchStateBooleanProperty.set(switchStateBooleanProperty);
+    }
+    /**
+     * returns valueProperty.
+     *
+     * @return BooleanProperty
+     */
+    public BooleanProperty valueProperty() {
+        return this.switchStateBooleanProperty;
+    }
+    //</editor-fold>
```

Comment:
I think we need more instance variables for some of the parameters of the constructors and also getters and setters.

Reply:
Or maybe not, most variables can be queried from the rectangle/switch/animation.